### DISCUSSION TITLE
Routing issue between a site-to-site connection

### DISCUSSION THREAD

**Discussion by hohomouse** [strongswan/strongswan#466] [Q&A] [answered]
Hi team,

I created a site-to-site connection like this:
Site A: 102.114.xxx.xxx (public IP), subnet within/behind this host: 10.5.0.1/24
Site B: 172.104.xxx.xxx (public IP), subnet within/behind this host: 10.6.0.1/24

**My ipsec.conf on Site-A looks like this:**

```
conn Site-A-to-Site-B
    type=tunnel
    authby=secret
    ike=aes256-sha256-sha1-modp4096-modp2048-modp1024,3des-sha256-sha1-modp2048-modp1024!
    esp=aes256-sha256-sha1,3des-sha256-sha1!

    leftid=Site-A
    left=%defaultroute
    leftsubnet=10.5.0.0/24
    #leftfirewall=yes

    rightid=Site-B
    right=172.104.xxx.xxx
    rightsubnet=10.6.0.0/24
    #rightfirewall=yes

    mobike=no
    auto=start
```

**My ipsec.conf on Site-B looks like this:**

```
conn Site-B-to-Site-A
    type=tunnel
    authby=secret
    ike=aes256-sha256-sha1-modp4096-modp2048-modp1024,3des-sha256-sha1-modp2048-modp1024!
    esp=aes256-sha256-sha1,3des-sha256-sha1!

    leftid=Site-B
    left=%defaultroute
    leftsubnet=10.6.0.0/24
    #leftfirewall=yes

    rightid=Site-A
    right=102.114.xxx.xxx
    rightsubnet=10.5.0.0/24
    #rightfirewall=yes

    mobike=no
    auto=start
```

Strongswan started successfully on both sites and I saw SA was up (only paste Site A here):
```
Security Associations (1 up, 0 connecting):
Site-A-to-Site-B[2]: ESTABLISHED 18 minutes ago, 172.104.xxx.xxx[Site-A]...102.114.xxx.xxx[Site-B]
Site-A-to-Site-B[2]: IKEv2 SPIs: fc506fe13948e896_i f233aeaa3ffcc246_r*, pre-shared key reauthentication in 2 hours
Site-A-to-Site-B[2]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_4096
Site-A-to-Site-B{2}:  INSTALLED, TUNNEL, reqid 2, ESP in UDP SPIs: c79cfe7e_i c2aad8b6_o, IPCOMP CPIs: 3432_i 9666_o
Site-A-to-Site-B{2}:  AES_CBC_256/HMAC_SHA2_256_128, 0 bytes_i, 0 bytes_o, rekeying in 2 hours
Site-A-to-Site-B{2}:   10.5.0.0/24 === 10.6.0.0/24
```

Also, I saw in the log the route was added, but I did it see it exists in the system routing table.
Here are some logs containing routing added (but no found):

```
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> adding policy 10.5.0.0/24 === 10.6.0.0/24 in [priority 375423, refcount 1]
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> adding policy 10.5.0.0/24 === 10.6.0.0/24 fwd [priority 375423, refcount 1]
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> adding policy 10.6.0.0/24 === 10.5.0.0/24 out [priority 375423, refcount 1]
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> getting a local address in traffic selector 10.6.0.0/24
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> using host 10.6.0.1
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> getting iface name for index 2
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> using 172.104.xxx.xxx as nexthop and eth0 as dev to reach 104.112.xxx.xxx/32
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> installing route: 10.5.0.0/24 via 172.104.xxx.xxx src 10.6.0.1 dev eth0
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> getting iface index for eth0
Jun 20 22:46:14 12[IKE] <Site-A-to-Site-B|1> CHILD_SA Site-A-to-Site-B{1} established with SPIs ccee8b67_i c52f056b_o and TS 10.6.0.0/24 === 10.5.0.0/24
Jun 20 22:46:14 12[CHD] <Site-A-to-Site-B|1> CHILD_SA Site-A-to-Site-B{1} state change: INSTALLING => INSTALLED
Jun 20 22:46:14 12[KNL] <Site-A-to-Site-B|1> 172.104.xxx.xxx is on interface eth0
Jun 20 22:46:14 12[IKE] <Site-A-to-Site-B|1> received AUTH_LIFETIME of 10048s, scheduling reauthentication in 9508s
Jun 20 22:46:14 12[IKE] <Site-A-to-Site-B|1> activating new tasks
Jun 20 22:46:14 12[IKE] <Site-A-to-Site-B|1> nothing to initiate
Jun 20 22:46:54 08[KNL] <Site-A-to-Site-B|1> querying SAD entry with SPI ccee8b67
Jun 20 22:46:54 08[KNL] <Site-A-to-Site-B|1> querying SAD entry with SPI c52f056b
```

What I want is from Site-A host, e.g. 10.5.0.1, to be able to ping a host in Site-B, e.g. 10.6.0.1.
My knowledge is not enough to troubleshoot or even I misunderstand it.

Appreciate your help. Thanks a lot.

Regards,
HoHo

**Comment by tobiasbrunner** [comment, marked as answer]
> Also, I saw in the log the route was added, but I did it see it exists in the system routing table.

Routes are installed in routing table 220 (`ip route list table 220`).

**Reply by hohomouse** [reply]
Hi @tobiasbrunner, 

Thanks for quick reply! Yes, it's on table 220.

```
$ ip route list table 220
10.6.0.0/24 via 172.17.15.xxx dev eth0 proto static src 10.5.0.1 
```
But seems there are other problems preventing ping from one site to another.

e.g. I tried `ping 10.6.0.1` from 10.5.0.1 or on the strongswan host itself, and then I used tcpdump to listen ICMP on 10.6.0.1, but not traffic found.

I have forwarding enabled on both strongswan hosts.
```
net.ipv4.ip_forward=1
net.ipv4.conf.all.accept_redirects = 0
net.ipv4.conf.all.send_redirects = 0
net.ipv6.conf.all.forwarding=1
```

Is table 220 routing is in used by default routing?
Or where do you think the problem is?

Thanks,
HoHo

**Reply by tobiasbrunner** [reply]
> I have forwarding enabled on both strongswan hosts.

Forwarding is not necessary to access IPs on the VPN gateways themselves. It will be if you access hosts behind the gateways later.

> Is table 220 routing is in used by default routing?

strongSwan adds a rule (check `ip rule`) for it so routes in that table are preferred over those in the main routing table. But the route only matters for the source IP selection (i.e. so a ping to `10.6.0.1` is sent from `10.5.0.1` automatically).

Is a packet actually sent, is one received? (Check the packet counters in the status output on both sides.)

Could also be a firewall issue. Please post the output of `iptables-save` from both hosts.

**Reply by hohomouse** [reply]
Hi @tobiasbrunner ,

Here's the result of `ip rule`:
```
[root@Site-A ~]$ ip rule
0:	from all lookup local 
220:	from all lookup 220 
32766:	from all lookup main 
32767:	from all lookup default 
```
Yes, it looks up at table 220 before main. So the rule listed below should be checked.
```
$ ip route list table 220
10.6.0.0/24 via 172.17.15.xxx dev eth0 proto static src 10.5.0.1 
```
The IP 172.17.15.xxx is the default gateway for my strongswan host.

And I think the ICMP packets are being sent, below the eth0 is the default/public interface.
As you see the source is from 172.17.0.1, this is the internal IP address of my Site-A host. 
(Site A: 102.114.xxx.xxx (public IP), subnet within/behind this host: 10.5.0.1/24).
So it's NATed, but the packets should be sent out.
```
[root@Site-A ~]$ tcpdump -nnvv -i eth0 host 10.6.0.1
tcpdump: listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
20:31:31.865950 IP (tos 0x0, ttl 64, id 27713, offset 0, flags [DF], proto ICMP (1), length 84)
    172.17.0.1 > 10.6.0.1: ICMP echo request, id 7843, seq 308, length 64
20:31:32.865988 IP (tos 0x0, ttl 64, id 28709, offset 0, flags [DF], proto ICMP (1), length 84)
    172.17.0.1 > 10.6.0.1: ICMP echo request, id 7843, seq 309, length 64
```

I don't think it's firewall issue, as the OUTPUT chain is all ACCEPT.
I paste out the majority part of the `iptables-save`, thanks for reviewing.
Noticing there's a `-A POSTROUTING -s 10.5.0.0/24 ! -o docker0 -j MASQUERADE`,
but I don't think this impacts the packet sent from the VPN gateway (Above tcpdump, source ip is 172.17.0.1)
```
[root@Site-A ~]$ iptables-save 
# Generated by iptables-save v1.4.21 on Tue Jun 22 20:35:21 2021
*nat
:PREROUTING ACCEPT [93553:5849208]
:INPUT ACCEPT [37650:2990476]
:OUTPUT ACCEPT [44084:3444607]
:POSTROUTING ACCEPT [44079:3444187]
:DOCKER - [0:0]
-A PREROUTING -m addrtype --dst-type LOCAL -j DOCKER
-A OUTPUT ! -d 127.0.0.0/8 -m addrtype --dst-type LOCAL -j DOCKER
-A POSTROUTING -s 10.5.0.0/24 ! -o docker0 -j MASQUERADE
-A DOCKER -i docker0 -j RETURN
COMMIT
# Completed on Tue Jun 22 20:35:21 2021
# Generated by iptables-save v1.4.21 on Tue Jun 22 20:35:21 2021
*filter
:INPUT ACCEPT [0:0]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [458617:2822349903]
:DOCKER - [0:0]
:DOCKER-ISOLATION - [0:0]
-A INPUT -s 65.78.0.0/16 -p udp -j DROP
-A INPUT -s 60.224.0.0/16 -p udp -j DROP
-A INPUT -m state --state RELATED,ESTABLISHED -j ACCEPT
-A INPUT -p icmp -j ACCEPT
-A INPUT -i lo -j ACCEPT
...
... Some other INPUT ACCEPT rules
...
-A INPUT -j REJECT --reject-with icmp-host-prohibited
-A FORWARD -j DOCKER-ISOLATION
-A FORWARD -o docker0 -j DOCKER
-A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i docker0 ! -o docker0 -j ACCEPT
-A FORWARD -i docker0 -o docker0 -j ACCEPT
-A FORWARD -j REJECT --reject-with icmp-host-prohibited
-A DOCKER-ISOLATION -j RETURN
COMMIT
# Completed on Tue Jun 22 20:35:21 2021
```

Sorry I don't find a way to counter ICMP packets sent out from eth0.

**Reply by tobiasbrunner** [reply]
> And I think the ICMP packets are being sent, below the eth0 is the default/public interface.
> As you see the source is from 172.17.0.1, this is the internal IP address of my Site-A host.
> (Site A: 102.114.xxx.xxx (public IP), subnet within/behind this host: 10.5.0.1/24).
> So it's NATed, but the packets should be sent out.

That's wrong. You want to avoid the NAT for these packets, otherwise they don't match the IPsec policies, which cover the internal subnets not the natted addresses. So maybe add something like this:

    iptables -t nat -I POSTROUTING -s 10.5.0.0/24 -m policy --dir out --pol ipsec -j ACCEPT

Or even more generically:

    iptables -t nat -I POSTROUTING -m policy --pol ipsec --dir out -j ACCEPT

That way packets that match an IPsec policy won't get natted.

**Reply by hohomouse** [reply]
That's great! @tobiasbrunner. It works! At Site-B gateway, it receives the ICMP from Site-A.
Now I am working on how to return the echo request at Site-B.

Site-B has no NAT, eth0 is bound to the public IP address, but the 10.6.0.0/24 is on the interface docker0.
Now the problem is, eth0 receives the packets, but not forwarded to docker0.

From below `iptables-save` output for FOWARD chain, not sure what rule is missing.
```
-A FORWARD -s 10.5.0.0/24 -d 10.6.0.0/24 -i eth0 -m policy --dir in --pol ipsec --reqid 3 --proto esp -j ACCEPT
-A FORWARD -s 10.6.0.0/24 -d 10.5.0.0/24 -o eth0 -m policy --dir out --pol ipsec --reqid 3 --proto esp -j ACCEPT
-A FORWARD -s 10.5.0.0/24 -d 10.6.0.0/24 -i eth0 -m policy --dir in --pol ipsec --reqid 1 --proto esp -j ACCEPT
-A FORWARD -s 10.6.0.0/24 -d 10.5.0.0/24 -o eth0 -m policy --dir out --pol ipsec --reqid 1 --proto esp -j ACCEPT
-A FORWARD -j DOCKER-ISOLATION
-A FORWARD -o docker0 -j DOCKER
-A FORWARD -o docker0 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i docker0 ! -o docker0 -j ACCEPT
-A FORWARD -i docker0 -o docker0 -j ACCEPT
-A FORWARD -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i lo -j ACCEPT
-A FORWARD -j FORWARD_direct
-A FORWARD -j FORWARD_IN_ZONES_SOURCE
-A FORWARD -j FORWARD_IN_ZONES
-A FORWARD -j FORWARD_OUT_ZONES_SOURCE
-A FORWARD -j FORWARD_OUT_ZONES
-A FORWARD -m conntrack --ctstate INVALID -j DROP
-A FORWARD -j REJECT --reject-with icmp-host-prohibited
```

**Reply by tobiasbrunner** [reply]
`leftfirewall=yes` apparently already installed rules in that chain. Are you sure there is no NAT? Docker usually installs NAT rules for its subnets.

**Reply by hohomouse** [reply]
Hi @tobiasbrunner ,

I checked a few times, don't see anything suspicious on the NAT chain. 

Here's the output from eth0 (the public IP interface) at Site-B.
```
[root@Site-B ~]$ tcpdump -nnvv -i eth0 host 10.6.0.1
tcpdump: listening on eth0, link-type EN10MB (Ethernet), capture size 262144 bytes
23:36:09.276054 IP (tos 0x0, ttl 64, id 34632, offset 0, flags [DF], proto ICMP (1), length 84)
    10.5.0.1 > 10.6.0.1: ICMP echo request, id 9361, seq 29, length 64
23:36:10.276194 IP (tos 0x0, ttl 64, id 35147, offset 0, flags [DF], proto ICMP (1), length 84)
    10.5.0.1 > 10.6.0.1: ICMP echo request, id 9361, seq 30, length 64
```

No matter I use `leftfirewall=yes` or NOT at Site-B, I got the same tcpdump output as above.

But these packets are not found on the docker0 interface.  10.6.0.1 is the docker0 bridge IP. I can not think of any NAT rule that could make a difference on the routing to hit on docker0 interface. 

Also, I try to understand more by looking at this doc: https://rlworkman.net/howtos/iptables/chunkyhtml/c962.html, sorry I knowledge is not enough to get it right yet.

Thanks,
HoHo

**Reply by tobiasbrunner** [reply]
Ah, so it's an IP on the interface itself and not that of a container? In that case FORWARD is not relevant as it's a local IP. Check the INPUT and OUTPUT chains (with `leftfirewall=yes` make sure to enable `lefthostaccess=yes` too so rules are added to those two chains as well).

**Reply by hohomouse** [reply]
Hi @tobiasbrunner ,

I tried to use this tool (https://github.com/commonism/iptables-trace) to capture where have the packets been to.
```
[root@Site-B ~]$ python3 iptables-trace.py --source 10.5.0.1
Namespace(bpf='', chain=['OUTPUT', 'PREROUTING'], clear_chain=False, destination=None, limit=False, protocol=None, source='10.5.0.1', xmark_mask='0x800001ff')
IN=eth0 OUT= SRC=10.5.0.1 DST=10.6.0.1 LEN=84 TOS=0x00 PREC=0x00 TTL=64 ID=40630 PROTO=ICMP TYPE=8 CODE=0 
	raw PREROUTING NFMARK=0x0 (0x6a94d800)
		ACCEPT
	mangle PREROUTING (#1) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PREROUTING_direct 
	mangle PREROUTING_direct (#1) NFMARK=0x0 (0x6a94d800)
		return
	mangle PREROUTING (#2) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PREROUTING_ZONES_SOURCE 
	mangle PREROUTING_ZONES_SOURCE (#3) NFMARK=0x0 (0x6a94d800)
		return
	mangle PREROUTING (#3) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PREROUTING_ZONES 
	mangle PREROUTING_ZONES (#1) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PRE_public 
	mangle PRE_public (#1) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PRE_public_log 
	mangle PRE_public_log (#1) NFMARK=0x0 (0x6a94d800)
		return
	mangle PRE_public (#2) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PRE_public_deny 
	mangle PRE_public_deny (#1) NFMARK=0x0 (0x6a94d800)
		return
	mangle PRE_public (#3) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PRE_public_allow 
	mangle PRE_public_allow (#1) NFMARK=0x0 (0x6a94d800)
		return
	mangle PRE_public (#4) NFMARK=0x0 (0x6a94d800)
		return
	mangle PREROUTING NFMARK=0x0 (0x6a94d800)
		ACCEPT
	nat PREROUTING (#1) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PREROUTING_direct 
	nat PREROUTING_direct (#1) NFMARK=0x0 (0x6a94d800)
		return
	nat PREROUTING (#2) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PREROUTING_ZONES_SOURCE 
	nat PREROUTING_ZONES_SOURCE (#3) NFMARK=0x0 (0x6a94d800)
		return
	nat PREROUTING (#3) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PREROUTING_ZONES 
	nat PREROUTING_ZONES (#1) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PRE_public 
	nat PRE_public (#1) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PRE_public_log 
	nat PRE_public_log (#1) NFMARK=0x0 (0x6a94d800)
		return
	nat PRE_public (#2) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PRE_public_deny 
	nat PRE_public_deny (#1) NFMARK=0x0 (0x6a94d800)
		return
	nat PRE_public (#3) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> PRE_public_allow 
	nat PRE_public_allow (#1) NFMARK=0x0 (0x6a94d800)
		return
	nat PRE_public (#4) NFMARK=0x0 (0x6a94d800)
		return
	nat PREROUTING (#4) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 addrtype:{'dst-type': 'LOCAL'} 
		=> DOCKER 
	nat DOCKER (#2) NFMARK=0x0 (0x6a94d800)
		return
	nat PREROUTING NFMARK=0x0 (0x6a94d800)
		ACCEPT
	mangle INPUT (#1) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> INPUT_direct 
	mangle INPUT_direct (#1) NFMARK=0x0 (0x6a94d800)
		return
	mangle INPUT NFMARK=0x0 (0x6a94d800)
		ACCEPT
	filter INPUT (#2) NFMARK=0x0 (0x6a94d800)
		ip 10.5.0.0/255.255.255.0 -> 10.6.0.0/255.255.255.0 policy:{'dir': 'in', 'dir pol': 'ipsec', 'dir pol reqid': '4', 'dir pol reqid proto': 'esp'} 
		=> ACCEPT 
	security INPUT (#1) NFMARK=0x0 (0x6a94d800)
		ip 0.0.0.0/0.0.0.0 -> 0.0.0.0/0.0.0.0 
		=> INPUT_direct 
	security INPUT_direct (#1) NFMARK=0x0 (0x6a94d800)
		return
	security INPUT NFMARK=0x0 (0x6a94d800)
		ACCEPT
	nat INPUT NFMARK=0x0 (0x6a94d800)
		ACCEPT
```
Looks like there's no blocking for these ICMP packets to go from 10.5.0.1(eth0) to 10.6.0.1(docker0).
Even, I don't see blocking in the FORWARD chain, as I tried this:
```
iptables -I FORWARD -s 10.5.0.0/24 -d 10.6.0.0/24 -i eth0 -j ACCEPT
```
Please help take a look if there' anything wrong. 

Thanks,
HoHo

**Reply by tobiasbrunner** [reply]
As I said before, to access `10.6.0.1`, FORWARD is irrelevant (it's a local address, so it goes through INPUT) - it will only be to access other addresses in `10.6.0.0/24` that are not local. And yes, it doesn't seem to get blocked. Did you check what happens to the response?

**Reply by hohomouse** [reply]
Hi @tobiasbrunner,

I noticed something is different today, although the ipsec.conf(s) were the same as before.

From Site-A(10.5.0.1), ping Site-B(10.6.0.1), when the SA is established, even on the eth0 of Site-A VPN gateway, using `tcpdump -nnvv -i eth0 host 10.6.0.1` CAN NOT see any packet. (This was NOT the case 2 days ago).

```
[root@Site-A ~]$ ip route list table 220
10.6.0.0/24 via 172.17.15.xxx dev eth0 proto static src 10.5.0.1 
```
If I remove the routing in table 220, then tcpdump can see the ICMP packets again.

If that's the "right" case, then when the ICMP packets arrive at Site-B (eth0), tcpdump will see nothing if using `tcpdump -nnvv -i eth0 host 10.6.0.1`.

I am confused if this is the right case, seem like the ICMP packets that go into "an encrypted channel" can not be found by tcpdump.

Regards,
HoHo

**Reply by tobiasbrunner** [reply]
> I am confused if this is the right case, seem like the ICMP packets that go into "an encrypted channel" can not be found by tcpdump.

They can't, see [this FAQ entry](https://wiki.strongswan.org/projects/strongswan/wiki/FAQ#Capturing-outbound-plaintext-packets-with-tcpdumpwireshark).

**Reply by hohomouse** [reply]
Hi @tobiasbrunner , 

Now it works!. I have to add the same iptables rule at Site-B to avoid packets being NATed by default docker rules.

So, there's a final question. That is, in fact, using the docker bridge IP as the ping final destination may not be a good idea. For example, Site-A 10.5.0.1 ping Site-B 10.6.0.1 (bridge IP), now I can see the ping returns, but using `tcpdump -nnvv -i docker0` will see NOTHING. So my question is, why tcpdump can not see the ICMP packets hitting the docker0 interface when the packets are actually getting responded?

**Reply by tobiasbrunner** [reply]
Because the traffic never actually goes through that interface. Linux doesn't really care on which interface a local IP address is, i.e. there is no routing involved nor does traffic have to hit that interface to respond from that IP.

**Reply by hohomouse** [reply]
Hi @tobiasbrunner, thank you so much for your help and enlightening explanation. Have a nice weekend. HoHo
